web/tests/libtest/codechecker.py

```diff
@@ -224,7 +231,76 @@ def check(codechecker_cfg, test_project_name, test_project_path):
         return cerr.returncode
 
 
-def analyze(codechecker_cfg, test_project_name, test_project_path):
+def log(codechecker_cfg, test_project_path, clean_project=False):
+    """Log a test project."""
+    build_cmd = project.get_build_cmd(test_project_path)
+    build_json = os.path.join(codechecker_cfg['workspace'], "build.json")
+
+    if clean_project:
+        ret = project.clean(test_project_path)
+        if ret:
+            return ret
+
+    log_cmd = ['CodeChecker', 'log',
+               '-o', build_json,
+               '-b', "'" + build_cmd + "'",
+               ]
+
+    try:
+        proc = subprocess.Popen(shlex.split(' '.join(log_cmd)),
+                                cwd=test_project_path,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                env=codechecker_cfg['check_env'])
+        out, err = proc.communicate()
+        print(out)
+        print(err)
+
+        return 0
+    except CalledProcessError as cerr:
+        print("Failed to call:\n" + ' '.join(cerr.cmd))
+        return cerr.returncode
+
+
+def analyze(codechecker_cfg, test_project_path):
+    """Analyze a test project.
+
+    A build.json file should be in the workspace directory!
+    """
+    build_json = os.path.join(codechecker_cfg['workspace'], "build.json")
+
+    analyze_cmd = ['CodeChecker', 'analyze',
+                   build_json,
+                   '-o', codechecker_cfg['reportdir'],
+                   '--analyzers', 'clangsa'
```

Comment:
Why only `clangsa` is supported?

Reply:
For most of the tests we run `clangsa` only to analyze the source files, if we would enable clang-tidy for every test it would produce more results and all the result counts in the tests should be updated.
There are only a few tests where clang-tidy is enabled.